odoo/addons/base/models/ir_model.py

```diff
@@ -937,6 +918,163 @@ def _add_manual_fields(self, model):
                     model._add_field(name, field)
 
 
+class IrModelSelection(models.Model):
+    _name = 'ir.model.fields.selection'
+    _order = 'sequence, id'
+    _description = "Fields Selection"
+
+    field_id = fields.Many2one("ir.model.fields",
+        required=True, ondelete="cascade",
+        domain=[('ttype', 'in', ['selection', 'reference'])])
+    value = fields.Char(required=True)
+    name = fields.Char(translate=True, required=True)
+    sequence = fields.Integer(default=10)
+
+    _sql_constraints = [
+        ('selection_field_uniq', 'unique(field_id, value)',
+         'Selections values must be unique per field'),
+    ]
+
+    @tools.ormcache('model_name')
+    def _existing_field_data(self, model_name):
+        """ Return the given model's existing field data. """
+        cr = self._cr
+        cr.execute("""
+            SELECT f.name as field_name, s.value, s.name, s.sequence
+            FROM ir_model_fields f
+            JOIN ir_model_fields_selection s
+            ON f.id = s.field_id
+            WHERE model=%s
+            ORDER BY s.sequence, s.id
+            """, [model_name])
+        return {row['field_name']: row for row in cr.dictfetchall()}
+
+    @tools.ormcache()
+    def _get_field_data(self, field_id):
+        """ Return the given field's selection. """
+        cr = self._cr
+        cr.execute("""
+            SELECT value, name
+            FROM ir_model_fields_selection
+            WHERE field_id=%s
+            ORDER BY sequence, id
+        """, (field_id,))
+        return [(res[0], res[1]) for res in cr.fetchall()]
+
+    def _reflect_field_params(self, field):
+        if field.type not in ('selection', 'reference') or not isinstance(field.selection, list):
+            return []
+
+        field_id = self.env['ir.model.fields']._get_id(field.model_name, field.name)
+        if not field_id:
+            _logger.error("No field_id found for %s", field)
+            return []
+
+        result = []
+        for sequence, selection in enumerate(field.selection, start=10):
+            result.append({
+                'field_id': field_id,
+                'value': str(selection[0]),  # value may be an int
+                'name': selection[1],
+                'sequence': sequence,
+                'field_name': field.name,
```

Comment:
Not exactly, it is just to make sure when I compare old_vals and new_vals that I get the same keys. I get field_name in the old_vals because of the SQL query result.

Reply:
You may post-process SQL results to avoid that key in `old_vals`...